src/effects/VST3/VST3Wrapper.cpp

```diff
@@ -544,28 +552,61 @@ void VST3Wrapper::StoreSettings(EffectSettings& settings) const
    std::swap(vst3settings, GetSettings(settings));
 }
 
-bool VST3Wrapper::LoadPreset(Steinberg::IBStream* fileStream)
+void VST3Wrapper::LoadPreset(const wxString& presetId)
+{
+   using namespace Steinberg;
+
+   auto fileStream = owned(Vst::FileStream::open(presetId.c_str(), "rb"));
+   if(!fileStream)
+      throw FileException(FileException::Cause::Open, presetId);
+
+   if(!LoadPresetFromStream(fileStream))
+   {
+      throw SimpleMessageBoxException(
+         ExceptionType::Internal,
```

Comment:
Is `Internal` the right type for this and the other new exception, not `BadEnvironment`?  If this failure occurs, do we want Sentry data about it or not?

`FileException` does use `Internal` but see the comment in its constructor explaining that as an unusual decision.

Reply:
> Is Internal the right type for this and the other new exception, not BadEnvironment? If this failure occurs, do we want Sentry data about it or not?

Not sure how to answer the first question, but I don't think that it's a good idea to send preset loading failures to sentry. `LoadPresetFromStream(fileStream)` load preset file from the disc, and user may attempt to load preset that was genrated by older version of same plugin, or even by another plugin.

Reply:
If there should not be Sentry reporting for this error, then Internal is the wrong choice.  See this other example:

```
         GuardedCall(
            [&message, rc] {
            throw SimpleMessageBoxException{ rc != SQLITE_FULL ? ExceptionType::Internal : ExceptionType::BadEnvironment,
               message, XO("Warning"), "Error:_Disk_full_or_not_writable" }; },
            SimpleGuard<void>{},
            [this](AudacityException * e) {
               // This executes in the main thread.
               if (mCallback)
                  mCallback();
               if (e)
                  e->DelayedHandlerAction();
            }
         );
```

If Sqlite fails but tells the program it's exhaustion of device space, then we don't need to collect a report.  Only if Sqlite fails for some other various reason do we want a report.